multistream-2.0/spec.md

```diff
@@ -0,0 +1,403 @@
+# Multistream 2.0
+
+This proposal describes a replacement protocol for multistream-select.
+
+## Protocols
+
+This document proposes 6 new, micro-protocols with two guiding principles:
+
+1. Composition over complexity.
+2. Every byte and round-trip counts.
+
+This document *does not*, in fact, propose a protocol *negotiation* protocol.
+Instead, it proposes a set of stream/protocol management protocols that can be
+composed to flexibly negotiate protocols.
+
+First, this document proposes 4 protocol "negotiation" protocols. "Negotiation"
+is in quotes because none of these protocols actually involve negotiating
+anything.
+
+1. `multistream/advertise`: Inform the remote end about which protocols we speak
+   and. This should partially replace the current identify protocol.
```

Comment:
incomplete sentence. and? 

Is this the equivalent of `identify/push`?

Reply:
Yes. Fixed.

Reply:
> Is this the equivalent of identify/push?

Yes except:

1. It's designed to be significantly more efficient.
2. It only covers protocol advertisement. I'd like to try to avoid monolithic protocols in the future as small protocols like this are easier to mix/match/upgrade.